### ISSUE TITLE
Which version of the MvcReportViewer is compatible with Microsoft SQL Server Reporting Services Version 10.50.6000.34 now?

### ISSUE THREAD

**Issue by mkamoski** [ilich/MvcReportViewer#80]
Ilich (and all) -- 

Please help.

I was able to browse the directory where my reports are stored on the server, using a web browser.

At the report listing page, directory browsing, at the bottom of the page it said the following...

Microsoft SQL Server Reporting Services Version 10.50.6000.34

...so does that help me to know which version of the MvcReportViewer will work for my reports?

Please advise.

Thanks.

-- Mark Kamoski

**Comment by ilich** [comment]
It seems that you are using SQL Server 2008 R2. I developed my MvcReportViewer using SQL Server 2012. This is the reason of using Report Viewer 11.0. I thought it has to be backward compatible, but it does not seem so. I'd recommend to take the source code, change Microsoft.ReportViewer.WebForms, Version=11.0.0.0 to be Version=10.0.0.0 and build. Then try the library in the application. I think it will work.

If it still does not work then try to build simple reporting application using ASP.NET Web Forms. If it works then you will have an idea what to fix in MvcReportViewer. If it does not work than you have to ask your DBA/IT Support to check what is going on with SSRS installation.

**Comment by mkamoski** [comment]
Ilich (and all) -- 

It turns out that the root cause was the dreaded "MissingMethodException Method not found: 'Boolean Microsoft.ReportingServices.Common.DateTimeUtil.TryParseDateTime" for all of my woes, or so it seems to me and I found a simple (but somewhat hack) way of getting past it.

It seems to me that there was some kind of version incompatibility issue, and I do not know exactly what is on the server, and I do not have control of what is on the server.

But the MissingMethodException issue was only expressed on reports that had DateTime parameters, hence the underlying call to TryParseDateTime. So, all I did was convert my parameters with data-type DateTime to be data-type text. I know it is a workaround and kind of a hack, but I used it anyway and it worked beautifully. The downside is that I do not get the nice popup calendar inputs on the report toolbar when it renders, but that is OK. Sure, if they put a non-date value in the textbox then the report will say something like "cannot convert input to valid date time" but it would have said that anyway if the user put in junk.

So problem solved, all is well, and thanks so much for all your help.

Thanks.

-- Mark Kamoski
